firebase-storage/src/test/java/com/google/firebase/storage/UploadTest.java

```diff
@@ -420,25 +364,19 @@ public void fileUploadWithPauseCancel() throws Exception {
 
     Task<StringBuilder> task =
         TestUploadHelper.fileUploadWithPauseCancel(factory.getSemaphore(), sourceFile);
+
     // This is 20 seconds due to a fairness bug where resumed tasks can be put at the end.
-    for (int i = 0; i < 20000; i++) {
-      Robolectric.flushForegroundThreadScheduler();
-      if (task.isComplete()) {
-        // success!
-        TestUtil.verifyTaskStateChanges("fileUploadWithPauseCancel", task.getResult().toString());
-        Assert.assertTrue(ResumableUploadCancelRequest.CANCEL_CALLED);
-        return;
-      }
-      Thread.sleep(1);
-    }
-    Assert.fail();
+    TestUtil.await(task, 20, TimeUnit.SECONDS);
```

Comment:
Yeah this ^. This non-default timeout is documented, so we know it's deliberate and meaningful. We should (eventually) do that for all of them. You could even enforce that by changing `TestUtil.await(Task, int, TimeUnit)` to `TestUtil.await(Task, int, TimeUnit, String reasonWhyWereNotUsingTheDefault)` (which is a little odd, but would be pretty effective.) Or, since I see this same reason is used again below, `TestUtil.awaitFairnessBug(task)`

(Aside: 20s timeout in a "unit" test is questionable.)

No action required.

Reply:
Acknowledged, but left the timeouts and the comment as is for now.